### PR TITLE
Turn some not-too-useful TODOs into useful documentation

### PR DESCRIPTION
**Pull request by bjoernQ**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [ ] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/CONTRIBUTING.md) and followed its instructions.

#### Description
Some TODOs turned out to be just noise

#### Testing
Nothing should have changed

### CODE REVIEW

File: esp-hal/src/clock/mod.rs [hunk 1]
```diff
@@ -274,7 +274,6 @@ pub struct Clocks<'d> {
     pub pll_48m_clock: HertzU32,
     #[cfg(esp32h2)]
     pub pll_96m_clock: HertzU32,
-    // TODO chip specific additional ones as needed
```
Comment (bjoernQ) [inline esp-hal/src/clock/mod.rs:277]: This is now a normal process during chip-bring-up

File: esp-hal/src/mcpwm/mod.rs [hunk 2]
```diff
@@ -22,7 +22,12 @@
 //!       registers with flexible updating methods.
 //! * Fault Detection Module (Not yet implemented)
 //! * Capture Module (Not yet implemented)
-//!
+#![doc = ""]
```
Comment (bjoernQ) [inline esp-hal/src/mcpwm/mod.rs:25]: alternative would be to document it for all chips - I guess having just the right information for the currently selected chip is better?

File: esp-hal/src/rtc_cntl/mod.rs [hunk 3]
```diff
@@ -647,7 +647,6 @@ impl RtcClock {
         (100_000_000 * 1000 / period) as u16
     }
 
-    // TODO: implement for ESP32-C6, and H2
```
Comment (bjoernQ) [inline esp-hal/src/rtc_cntl/mod.rs:650]: that's already done

File: esp-hal/src/mcpwm/mod.rs (hunk 2 above)
Comment (jessebraham) [reply esp-hal/src/mcpwm/mod.rs:25]: Yeah I've been wondering what we should do about this as well. I guess in this case documenting it on a per-chip basis is simple enough, so this should be fine.

**Review by jessebraham — APPROVED** [review]
LGTM, thanks!

File: esp-hal/src/mcpwm/mod.rs (hunk 2 above)
Comment (MabezDev) [reply esp-hal/src/mcpwm/mod.rs:25]: This is the nicest way to do it I've seen so far, so I'm going to steal this method :D. Thanks!

**Review by SergioGasquez — APPROVED** [review]
LGTM! Thanks

File: esp-hal/src/mcpwm/mod.rs (hunk 2 above)
Comment (MabezDev) [reply esp-hal/src/mcpwm/mod.rs:25]: FYI it doesn't add any newlines, so you can just inline the bit you want to change, for example:

```rust
//! The System Timer is a 
#![cfg_attr(esp32s2, doc = "`64-bit`")]
#![cfg_attr(not(esp32s2), doc = "`54-bit`")]
//! timer with three comparators capable of signalling an alarm interupt on each.
```
Creates

image

Which is nice.

**Pull request merged**